service/worker/archiver/archiver.go

```diff
@@ -123,7 +128,16 @@ func handleRequest(ctx workflow.Context, logger log.Logger, metricsClient metric
 	}
 	uploadSW.Stop()
 
-	if err != nil {
+	blobsToDelete := result.BlobsToDelete
+	if timeoutErr, ok := err.(*workflow.TimeoutError); ok {
```

Comment:
Can you comment this bit of code, it took me a little while to understand why this exists.

Reply:
Comments added.